updater/lib/dependabot/dependency_snapshot.rb

```diff
@@ -210,6 +218,8 @@ def parse_files!
 
     sig { returns(Dependabot::FileParsers::Base) }
     def dependency_file_parser
+      raise "Current directory not set" if @current_directory == ""
```

Comment:
can we raise `DependabotError` with a more specific message here so it is easier to track via Sentry?

Reply:
I refactored this to be more clear that this is an assertion, something we need to fix on our side if we see it. Raising a DependabotError now instead of a RuntimeError.